### ISSUE TITLE
iOS iDetection Speed Table

### ISSUE THREAD

**Issue by glenn-jocher** [ultralytics/yolov5#1276]
## 🚀 Feature
We'd like to create an iDetection Speed Table for each year's iPhone release, going back as far as iDetection will run, which may be around iPhone 5 or 6. Crowd-sourcing the table will help show real-world performance of the models on various hardware. 

**If you have an iOS device, you can contribute** by downloading [iDetection](https://apps.apple.com/us/app/idetection/id1452689527), testing your speeds, and then uploading your results here. We try to test near 100% battery at room temperature (about 25°C), and record the speed after about 10 seconds of inference for each model. 

Please help by uploading your results! Thank you!!

<img src="https://user-images.githubusercontent.com/26833433/98019191-be36e000-1e01-11eb-97e0-69ca25c30531.jpg" width="800">

## iDetection v7.8 Inference Speeds

&nbsp; | Year | ASIC<br>-process | ANE<br>(TOPS) || YOLOv5s<br>(ms)   |YOLOv5m<br>(ms)   |YOLOv5l<br>(ms)   |YOLOv5x<br>(ms)
--- |---  |---  |---  |-|---   |---   |---   |---
iPhone 6  | 2014  | [A8-20nm](https://en.wikipedia.org/wiki/Apple_A8)  |-  ||90  |180  |350  |500
iPhone 6s  | 2015  | [A9-16nm](https://en.wikipedia.org/wiki/Apple_A9)  |-  ||148  |216  |304  |475
iPhone 7  | 2016  | [A10-16nm](https://en.wikipedia.org/wiki/Apple_A10)  |-  ||94.7  |140.7  |216.7  |289.8
iPhone 8/X  | 2017  | [A11-10nm](https://en.wikipedia.org/wiki/Apple_A11)  |0.6  ||-  |-  |-  |-
iPhone XR/XS  | 2018  | [A12-7nm](https://en.wikipedia.org/wiki/Apple_A12)  |5.0  ||22.3  |25.8  |43.2 |57.7
iPhone 11 | 2019  | [A13-7nm](https://en.wikipedia.org/wiki/Apple_A13)  |6.0 ||17.4  |21.3  |27.8  |41.2
iPhone 12 | 2020  | [A14-5nm](https://en.wikipedia.org/wiki/Apple_A14)  |11.0  ||14.3  |16.5  |21.0  |28.8
iPhone 13 | 2021  | [A15-5nm](https://en.wikipedia.org/wiki/Apple_A15)  |15.8  ||  |  |  |

*CoreML models exported as FP8 320x192 with [release v3.1](https://github.com/ultralytics/yolov5/releases)
*Measured with [iDetection v7.8](https://apps.apple.com/us/app/idetection/id1452689527) at 100% battery at 25°C. Average speed after 10 seconds recorded.

**Comment by NanoCode012** [comment]
Hi Glenn, I got an old 7Plus. I think the results will definitely be lower than a new one.

5s - 10.4 FPS - 94.7 ms
5m - 7.1 FPS - 140.7 ms
5l - 4.6 FPS - 216.7 ms
5x - 3.7 FPS - 289.8 ms

Started from 100%. Default Params.

**Comment by glenn-jocher** [comment]
@NanoCode012 great, thanks bud! I've added your data points to the table.

**Comment by glenn-jocher** [comment]
Here are the results of a recent run for iDetection v7.8 YOLOv5s iPhone 11 for 20 minutes, starting from 94% battery at room temperature. The x axis is frame count. About 36000 frames were processed over 1200s (20 minutes), equal to about 30 FPS. You can see initial speed of 17 ms matches table numbers for this device, with thermal throttling starting at about 10 minutes, with iOS automatically reducing processing capacity to about 60% original, resulting in inference times of 27 ms. I don't track memory or thermal state, I'll try to add them in the future. But according to the below charts, everything seems to be working well. It's also possible that longer runs will result in more throttling, so I'll try a longer run to 60 minutes in the future also.

![results](https://user-images.githubusercontent.com/26833433/102024693-68cae880-3d48-11eb-8a36-c6d2f820f9d1.png)

**Comment by glenn-jocher** [comment]
Additional profiling information for iDetection v7.8 YOLOv5s iPhone 11 run. Most importantly, RAM usage is steady over time and very low at about 100 MB out of 4 GB.

## RAM Usage
![Screen Shot 2020-12-13 at 2 35 49 PM](https://user-images.githubusercontent.com/26833433/102026070-c4996f80-3d50-11eb-8967-335ef8887e40.jpg)

## CPU Usage
![Screen Shot 2020-12-13 at 2 35 46 PM](https://user-images.githubusercontent.com/26833433/102026066-c2cfac00-3d50-11eb-9e9d-311c7ee44a66.jpg)

## Energy Impact
![Screen Shot 2020-12-13 at 2 35 51 PM](https://user-images.githubusercontent.com/26833433/102026073-c5320600-3d50-11eb-8981-642d0e296839.jpg)

**Comment by glenn-jocher** [comment]
Updated results for iDetection v7.8 YOLOv5s iPhone 11 for 60 minutes. Plots include system RAM usage over time now, indicating very small app memory usage (only 70 MB! :). Conclusion: iPhone 11 is able to run YOLOv5s steady state in realtime 30 FPS, even after thermal throttling begins around 10 minutes into usage.

![results](https://user-images.githubusercontent.com/26833433/102128273-793f9980-3e02-11eb-9dc4-dc0fda820bb2.png)

**Comment by glenn-jocher** [comment]
Updated results for iDetection v7.8 YOLOv5s and YOLOv5m on iPhone 11 for **60 minutes** of continuous usage (about 120,000 images processed each). Plots include system RAM usage over time, indicating very small app memory usage (only 70 MB for YOLOv5s and 80 MB for YOLOv5m! :). Conclusion: iPhone 11 is able to run YOLOv5s/m steady state at or near  realtime 30 FPS, even after thermal throttling begins around 15 minutes into usage. Results are excellent. Plan to reproduce same test on iPhone 12 soon.

![results](https://user-images.githubusercontent.com/26833433/102246206-56b98900-3eb3-11eb-831e-7d37a615591b.png)

EDIT: Raw data uploaded below for above plots.
[frames_iPhone11_YOLOv5s.txt](https://github.com/ultralytics/yolov5/files/5699947/frames_iPhone11_YOLOv5s.txt)
[frames_iPhone11_YOLOv5m.txt](https://github.com/ultralytics/yolov5/files/5699946/frames_iPhone11_YOLOv5m.txt)

**Comment by glenn-jocher** [comment]
iPhone 11 vs iPhone 12 results for iDetection v7.8 YOLOv5s **60 minutes** of continuous usage (>100,000 images processed each). Plots include system RAM usage over time, indicating very small app memory usage. Conclusion: iPhone 11 is able to run YOLOv5s/m steady state at or near  realtime 30 FPS, even after thermal throttling begins around 15 minutes into usage. **iPhone 12 is able to run YOLOv5s with zero throttling over 1 full hour of usage.** 

![idetection_profile](https://user-images.githubusercontent.com/26833433/102817629-fcbe3500-4384-11eb-82c2-1f6717b5284d.png)

EDIT: Raw data uploaded below for above plots.
[frames_iPhone11_YOLOv5s.txt](https://github.com/ultralytics/yolov5/files/5726296/frames_iPhone11_YOLOv5s.txt)
[frames_iPhone12_YOLOv5s.txt](https://github.com/ultralytics/yolov5/files/5726298/frames_iPhone12_YOLOv5s.txt)

**Comment by christli7** [comment]
Do you also need iPhone XR?
V5s 22 ms
V5m 25 ms
V5l 44 ms
V5x 57 ms

**Comment by glenn-jocher** [comment]
@christli7 thanks for the iPhone XR numbers! Since the XR and the XS share the same A12 ASICs we can rely on the XS numbers, which are almost exactly the same same as your XR numbers. I will update the table to note that the speeds for XS apply to XR.

**Comment by abfleishman** [comment]
old iPhone SE (2016?)
V5s 148ms
V5m 216ms
V5l 304ms
V5x 475ms

**Comment by glenn-jocher** [comment]
@abfleishman thanks for the profiling results! iPhone SE is on the A9 chipset, I will update the table with your values!

**Comment by jeannotes** [comment]
the results are confusing.
you said that:
 V5s 148ms V5m 216ms V5l 304ms V5x 475ms
thease numbers are times per image?
but you also mentioned that, 
iPhone 12	2020	A14-5nm	11.0	 four models runing time:14.3 16.5 21.0	28.8
what's the the meaning

**Comment by glenn-jocher** [comment]
@jeannotes table times are correct. Units are milliseconds per image in our iOS YOLOv5 app. See https://github.com/ultralytics/yolov5/issues/1276#issue-735508557

**Comment by jeannotes** [comment]
@glenn-jocher so how do you do that? any model compression?can you give us some toturials?

**Comment by fanatic-studio** [comment]
Where can I get the source code of iDetection, to test my own model

**Comment by glenn-jocher** [comment]
@view-team we have not open-sourced the iDetection source code, however soon you'll be able to preview your own custom models in iOS and Android using the Ultralytics HUB. Please visit https://ultralytics.com/hub to learn more.

**Comment by dubrovin-sudo** [comment]
> CoreML models exported as FP8 320x192

@glenn-jocher How to export a model with a resolution 320x192 if only one number needs to be specified in the argparse?

 For example, the default image size of 640 means 640x640:
`!python utils/benchmarks.py --weights yolov5n.pt --imgsz 640 --device 0`

**Comment by glenn-jocher** [comment]
@dubrovin-sudo `--imgsz 320 192`

**Comment by Marcof64** [comment]
Have tested YOLOv5 (m and x) and YOLOv8 (m and l) on iphone 13 Mini (A17, NPU rated 15 TOPS) vs. Iphone 15 Pro (A17 NPU rated 35 TOPS).

Suprisingly I get NO decrese in inference time!!!

How is it possibile?

YOLOv5m: 15 ms on both iPhone 13 and 15 Pro
YOLOv5x:  21 ms on both iPhone 13 and 15 Pro

YOLOv8m:

**Comment by Marcof64** [comment]
Have tested YOLOv5 (m and x) and YOLOv8 (m and l) on iphone 13 Mini (A17, NPU rated 15 TOPS) vs. Iphone 15 Pro (A17 NPU rated 35 TOPS).

Suprisingly I get NO decrese in inference time!!!

**How is it possibile?**

YOLOv5m: 15 ms on both iPhone 13 and 15 Pro
YOLOv5x: 21 ms on both iPhone 13 and 15 Pro

YOLOv8m: 14 ms on both iPhone 13 and 15 Pro
YOLOv8l:    21 ms on both iPHone 13 and 15 Pro


![Iphone 15 Pro YOLOv5m](https://github.com/ultralytics/yolov5/assets/104486700/20a9b7d7-c518-4b6d-8d14-2e23775b28c3)
![Iphone 15 Pro YOLOv5x](https://github.com/ultralytics/yolov5/assets/104486700/ff95173c-daf3-4635-b3f7-d55e89e1c8f2)
![iPhone 15 Pro YOLOv8l](https://github.com/ultralytics/yolov5/assets/104486700/d9e837db-1945-44dd-86d2-62b25601ac3d)
![iPhone 15 Pro YOLOv8m](https://github.com/ultralytics/yolov5/assets/104486700/d2dd1dee-a0af-43ac-b6ba-90f2d5f4849e)
![iPhone 13 mini YOLOv5m](https://github.com/ultralytics/yolov5/assets/104486700/30504b89-f05b-4034-814d-760b97bc61cc)
![iPhone 13 mini YOLOv5x](https://github.com/ultralytics/yolov5/assets/104486700/daffb33e-0b80-4885-b41c-4ac652ba69f5)
![iPhone 13 mini YOLOv8l](https://github.com/ultralytics/yolov5/assets/104486700/ef40572a-356c-4a1b-a7e7-86259b3e4600)
![iPhone 13 mini YOLOv8m](https://github.com/ultralytics/yolov5/assets/104486700/ffb4a9cf-c66b-486d-ab81-79d823f0c871)

**Comment by Marcof64** [comment]
Have tested YOLOv5 (m and x) and YOLOv8 (m and l) on iphone 13 Mini (A17, NPU rated 15 TOPS) vs. Iphone 15 Pro (A17 NPU rated 35 TOPS). 

Suprisingly I get NO decrese in inference time!!! **How is it possibile?**

The 15 Pro NPU has 2.3 the TOPS of the 13 mini NPU!

YOLOv5m: 15 ms on both iPhone 13 and 15 Pro
YOLOv5x: 21 ms on both iPhone 13 and 15 Pro

YOLOv8m: 14 ms on both iPhone 13 and 15 Pro
YOLOv8l: 21 ms on both iPHone 13 and 15 Pro

Iphone 13 mini screnshots:
![iPhone 13 mini YOLOv8m](https://github.com/ultralytics/yolov5/assets/104486700/409b628e-fade-428b-8e11-39967da4a644)
![iPhone 13 mini YOLOv5m](https://github.com/ultralytics/yolov5/assets/104486700/a1822fd7-e281-41db-8942-6c73442b07d8)
![iPhone 13 mini YOLOv5x](https://github.com/ultralytics/yolov5/assets/104486700/60539251-44de-4c41-9e68-5e4c3874c61b)
![iPhone 13 mini YOLOv8l](https://github.com/ultralytics/yolov5/assets/104486700/9569dd6a-9cdc-40fb-b689-716db338c1c8)

Iphone 15 Pro screenshots
![iPhone 15 Pro YOLOv8m](https://github.com/ultralytics/yolov5/assets/104486700/c3b7910f-4887-4a10-b5fb-8cb995c0b4af)
![Iphone 15 Pro YOLOv5m](https://github.com/ultralytics/yolov5/assets/104486700/766dcb80-c156-4237-ac7d-5443dee8e3fa)
![Iphone 15 Pro YOLOv5x](https://github.com/ultralytics/yolov5/assets/104486700/500c1bd3-9c43-438a-a4b9-19401bcc63a7)
![iPhone 15 Pro YOLOv8l](https://github.com/ultralytics/yolov5/assets/104486700/4b8cab3d-d57b-4976-a19d-e46e9e488f53)

**Comment by glenn-jocher** [comment]
@Marcof64 Interesting results! There can be various factors contributing to this, including software optimizations and other bottlenecks. We'll investigate the performance differences further. Thank you for sharing your findings!

**Comment by Marcof64** [comment]
Hi Glen,

 

Good to hear from you and congratulations with the Ultralytics growth and  YOLO VISION 22.

 

Let me know if I can be of any help with further test! I have both phones with me and access to more.

 

 

Side question: in the new “UltralyticsApp”, “640 imagesize” refers to the training resolution or the resolution being fed to YOLO for inference? What is the latter?

 

Thanks!!!

 

Marco

**Comment by glenn-jocher** [comment]
@Marcof64 Thank you for your kind words and for offering further testing! Your support is greatly appreciated. In the new UltralyticsApp, the "640 imagesize" refers to the resolution being fed to YOLO for inference. Thank you for your attention to detail and willingness to contribute!

**Comment by Marcof64** [comment]
Hello, I am getting strange results with YOLOv26, seems to be slower than YOLOv8 (on Iphone 17 Pro)

<img width="461" height="331" alt="Image" src="https://github.com/user-attachments/assets/b30e1b26-29e3-48e3-9990-068c8545a3b8" />



Also, does anybody have iOS performance of YOLO object tracking? How does it handle tracking more than 5 objects?

Thanks

**Comment by Marcof64** [comment]
Here is a more complete benchmark if useful:

<img width="2031" height="332" alt="Image" src="https://github.com/user-attachments/assets/a7a36dbf-0c09-46e8-92e9-f7622935df4d" />

**Comment by pderrenger** [comment]
Those app timings are usually end-to-end (camera/resize + model + NMS/render), and CoreML can route different models to different compute units (ANE vs GPU/CPU), so higher “TOPS” on paper won’t always reduce measured latency; see [Making Ultralytics YOLO Models Faster on Your Favorite Chip](https://www.ultralytics.com/blog/how-to-make-yolo-models-fast-on-your-favorite-chip) and the [Inference Latency guide](https://www.ultralytics.com/glossary/inference-latency). If you can share the per-stage breakdown (preprocess/inference/postprocess) plus the exact model variants, input size, precision, iOS version, and the CoreML compute-unit setting used by the app, we can tell if YOLO26 is spending more time in postprocess or falling back to CPU/GPU for some ops. For tracking, there’s no “5 objects” limit in principle, but speed mainly depends on how many detections you feed the tracker each frame (so filtering detections more aggressively and/or using a smaller model helps); see the [Ultralytics Docs on tracking](https://docs.ultralytics.com/modes/track/) for the general behavior.

**Comment by Marcof64** [comment]
Hello Paula,The test have been carried out using the Ultralytics hub Ultralytics yolo apps, Object recognition, coco set, standard settings, same iPhone 17 pro , same scene, you can probably replicate in your lab.Our target platform is smartphones, medium model, at least 30 FPSOn  iOS we are on target but surprisingly YOLO 26 is slower, on Android we get much lower than 30 FPS no matter what android phone is used. Have you managed to run on Android at at least 30 FPS a medium model on Coco recognition?By the way, we would be interested in FPS performance data on iOS when running object tracking on the Coco set. Do you by chance have the performance data?ThanksMarcoIl giorno 10 feb 2026, alle ore 11:26, Paula Derrenger ***@***.***> ha scritto:﻿pderrenger left a comment (ultralytics/yolov5#1276)
Those app timings are usually end-to-end (camera/resize + model + NMS/render), and CoreML can route different models to different compute units (ANE vs GPU/CPU), so higher “TOPS” on paper won’t always reduce measured latency; see Making Ultralytics YOLO Models Faster on Your Favorite Chip and the Inference Latency guide. If you can share the per-stage breakdown (preprocess/inference/postprocess) plus the exact model variants, input size, precision, iOS version, and the CoreML compute-unit setting used by the app, we can tell if YOLO26 is spending more time in postprocess or falling back to CPU/GPU for some ops. For tracking, there’s no “5 objects” limit in principle, but speed mainly depends on how many detections you feed the tracker each frame (so filtering detections more aggressively and/or using a smaller model helps); see the Ultralytics Docs on tracking for the general behavior.

**Comment by pderrenger** [comment]
If Android is “well under 30 FPS no matter what phone,” it almost always means you’re not actually hitting the device accelerator (NNAPI/GPU/NPU) and the runtime has fallen back to CPU, or you’re measuring end-to-end camera + resize + NMS/render rather than model-only; can you share the Android app log line that shows which backend/delegate is selected plus the exact model (e.g., yolo26m), precision (FP16/INT8), imgsz, and app version you tested in the Ultralytics Platform app? For a medium model on mobile, 30 FPS typically requires INT8 export plus a smaller inference input and tighter postprocess (e.g., higher `conf`, lower `max_det`) so NMS/tracking work doesn’t dominate; see the [Ultralytics Docs on export](https://docs.ultralytics.com/modes/export/) for the knobs to control precision and mobile formats. We don’t have a published iOS tracking FPS table to point to yet (tracking cost varies a lot by tracker/settings and how many detections you pass through), but if you post your tracking timings with the tracker name/config and `max_det/conf/imgsz`, I can help interpret what’s limiting you and we can use it as reference data here.

**Comment by pderrenger** [comment]
The Android Ultralytics app is intended to run via TensorFlow Lite with hardware delegates, but on Android it’s very common for the delegate (NNAPI/GPU/Hexagon) to be unavailable for a given device/model or to hit unsupported ops and silently fall back to CPU, which looks exactly like “slow on every flagship”; if the app exposes a Delegate/Accelerator setting, try switching it and then confirm what’s actually selected by grabbing an `adb logcat` snippet around model load (it should explicitly say which delegate is in use) and sharing that along with app version, Android version, model name, precision (FP16 vs INT8), and `imgsz`. Android can also lag iOS even with similar “TOPS” because accelerator support is fragmented and many NPUs are effectively INT8-first, so FP16 paths often end up on GPU/CPU while iOS CoreML more consistently routes to ANE; see the [Ultralytics TFLite export/deployment guide](https://docs.ultralytics.com/integrations/tflite/) and Google’s [TensorFlow Lite NNAPI delegate docs](https://android.googlesource.com/platform/external/tensorflow/+/refs/heads/upstream-master/tensorflow/lite/g3doc/android/delegates/nnapi.md) for what to look for in logs/settings.

**Comment by Marcof64** [comment]
Thanks Paula for the detailed answer.

 

Maybe there is an easier way to move ahead: we are using the “official” app that Ultralytics publishes on the GooglePlay app store, so your team knows if its app exposes a Delegate/Accelerator setting (we would in fact like to know), if it is switched and actually selected on the phones you have tested, you also know the models available in the app (we test with “medim”  model size), the precision used (FP16 vs INT8), and imgsz.

 

In other words, are there Android phones that in your tests do not fall to CPU or GPU when running the “official” Android App that Ultrlalytics publishes on the GooglePlay store?  If so, what models and OS version? We will then purchase the same phones, install the same OS versions and hopefully get at least 30 FPS 😊

 

Thanks

Marco

 

 

 

Does the

 

Thanks

Marco

**Comment by pderrenger** [comment]
Marco, we don’t maintain a “buy this Android phone/OS and the Play Store Ultralytics app will definitely use NPU” list because delegate availability is OEM-driver dependent and can change across Android updates, and the app may also run in an auto-select mode with no explicit toggle. The fastest way to move forward is to share a screenshot of the app’s Settings page plus the app version and the single `adb logcat` line(s) around model load so we can confirm whether it’s actually running on CPU/GPU/NNAPI; for a predictable 30 FPS “medium” model on Android you’ll typically need an INT8 TFLite plus a vendor NPU delegate (Qualcomm’s AI Engine Direct is the most straightforward path), rather than relying on generic NNAPI—see Google’s [LiteRT NPU delegate guide](https://ai.google.dev/edge/litert/next/npu). ([ai.google.dev](https://ai.google.dev/edge/litert/next/npu))
